### PR TITLE
[synthetics-command] handle and exit correctly on endpoints error

### PR DESCRIPTION
**Pull request by KevinYeramian**
### What and why?

- Handle and exit correctly on endpoints error: 
   - `getTest` 
   - `searchTests`
   -  `pollResults`
   -  `getPresignedURL`
   -  `triggerTests` 

### How?

Catch errors in try-catch! 

### Review checklist

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

File: src/commands/synthetics/run-test.ts [hunk 1]
```diff
@@ -44,24 +56,68 @@ export class RunTestCommand extends Command {
 
     const api = this.getApiHelper()
     const publicIdsFromCli = this.config.publicIds.map((id) => ({config: this.config.global, id}))
-    const testsToTrigger = publicIdsFromCli.length ? publicIdsFromCli : await this.getTestsList(api)
+    let testsToTrigger: {
+      config: ConfigOverride
+      id: string
+    }[]
```
Comment (m-rousse) [inline src/commands/synthetics/run-test.ts:62]:
```suggestion
    let testsToTrigger: TriggerConfig[]
```

File: src/commands/synthetics/run-test.ts [hunk 2]
```diff
@@ -44,24 +56,68 @@ export class RunTestCommand extends Command {
...
 
     if (!testsToTrigger.length) {
       this.reporter.log('No test suites to run.\n')
 
-      return 0
+      return safeExit(0)
     }
 
-    const {tests, overriddenTestsToTrigger, summary} = await getTestsToTrigger(api, testsToTrigger, this.reporter)
+    let testsToTriggerResult: {
```
Comment (m-rousse) [inline src/commands/synthetics/run-test.ts:90]:
```suggestion
    let testsToTrigger: {
```

File: src/commands/synthetics/__tests__/run-test.test.ts [hunk 3]
```diff
@@ -105,9 +107,148 @@ describe('run-test', () => {
...
+      const command = new RunTestCommand()
+      command.context = {stdout: {write}} as any
+      command['getApiHelper'] = (() => apiHelper) as any
+      command['publicIds'] = ['public-id-1', 'public-id-2']
+
+      expect(await command.execute()).toBe(1)
+    })
+
+    test('waitForResults throws', async () => {
+      console.log(utils.waitForResults)
```
Comment (m-rousse) [inline src/commands/synthetics/__tests__/run-test.test.ts:211]:
```suggestion
```

**Review by m-rousse — APPROVED** [review]
LGTM modulo 3 nits!

File: src/commands/synthetics/run-test.ts (hunk 2 above)
Comment (KevinYeramian) [reply src/commands/synthetics/run-test.ts:90]: `testsToTrigger` is already used

Comment (m-rousse) [reply src/commands/synthetics/run-test.ts:87]: Ok, all good in that case!

**Pull request merged**